tests/test_everflow_testbed.py

```diff
@@ -0,0 +1,664 @@
+        if bgp_peer['name'] == mg_facts['minigraph_neighbors'][dest_port]['name'] and ipaddress.IPAddress(bgp_peer['addr']).version == 4:
+            peer_ip = bgp_peer['addr']
+            break
+
+    return peer_ip, duthost.shell("ip neigh show %s | awk -F' ' '{print $5}'" %peer_ip)['stdout']
+
+
+class BaseEverflowTest(object):
+
+    @pytest.fixture(params=['tor', 'spine'])
+    def dest_port_type(self, request):
+        """
+        used to parametrized test cases on dest port type
+        :param request: pytest request object
+        :return: destination port type
+        """
+        return request.param
+
+    @pytest.fixture(scope='class')
+    def setup_mirror_session(self, duthost):
+        """Setup the Everflow Session"""
+
+        session_name =  "test_session_1"
+        session_src_ip =  "1.1.1.1"
+        session_dst_ip = "2.2.2.2"
+        session_ttl = "1"
+        session_dscp = "8"
+
+        session_prefixlens = ["24", "32"]
+        session_prefixes = []
+        for prefixlen in session_prefixlens:
+            session_prefixes.append(str(ipaddress.IPNetwork(session_dst_ip + "/" + prefixlen).network) + "/" + prefixlen)
+
+        duthost.command('config mirror_session add {} {} {} {} {}'.format(session_name, session_src_ip, session_dst_ip, session_dscp, session_ttl))
+
+        yield {'session_name' : session_name,
+               'session_src_ip' : session_src_ip,
+               'session_dst_ip' : session_dst_ip,
+               'session_ttl' : session_ttl,
+               'session_dscp' : session_dscp,
+               'session_prefixes': session_prefixes
+               }
+
+        duthost.command('config mirror_session remove {}'.format(session_name))
+
+
+    @abstractmethod
+    def setup_acl_table():
+        """
+        setup the acl table
+        return:pass
+        """
+        pass
+
+
+    @abstractmethod
+    def mirror_type():
+        """
+        used to parametrized test cases on mirror type
+        :param request: pytest request object
+        :return: mirror type
+        """
+        pass
+
+    @abstractmethod
+    def acl_stage():
+        """
+        get the acl stage
+        return:pass
+        """
+        pass
+
+
+    def test_everflow_case1(self, duthost, setup_info, setup_mirror_session, dest_port_type, partial_ptf_runner):
+        """  Test on Resolved route, unresolved route, best prefix match route creation and removal flows """
+
+        rx_port = setup_info[dest_port_type]['src_port']
+        rx_port_ptf_id =  setup_info[dest_port_type] ['src_port_ptf_id']
+        tx_port = setup_info[dest_port_type]['dest_port'][0]
+        tx_port_ptf_id = setup_info[dest_port_type]['dest_port_ptf_id'][0]
+        peer_ip, peer_mac = get_neighbor_info(duthost, tx_port)
+        
+        add_route(duthost, setup_mirror_session['session_prefixes'][0], peer_ip)
+
+        time.sleep(3)
+
+        partial_ptf_runner(setup_info, setup_mirror_session,self.acl_stage(), self.mirror_type(),
```

Comment:
Could you add some comments along the way describing what is going on here? It also might help to have a more descriptive name than "partial_ptf_runner" (unfortunately I'm still not sure what the partial is supposed to indicate so I don't have any good alternatives to suggest 😅).

Reply:
@daall Done

Reply:
I also found partial confusing and not clarified yet.

Is partial in ptf test? or is partial in part of this everflow test? The comments should be added to the function partial_ptf_runner.

Reply:
@yxieca Sure Added more description of the partial_ptf_runner fixture. It's part of everflow_test and call the library function ptf_runner.

Reply:
@yxieca LGTM, I'll let you take a look before resolving this